packages/eslint-plugin/tests/rules/key-spacing.test.ts

```diff
@@ -1317,5 +1350,87 @@ class Wacky {
         { messageId: 'missingValue' },
       ],
     },
+    {
+      code: `
+class Foo {
+  a:  (b)
+}
+      `,
+      output: `
+class Foo {
+  a: (b)
+}
+      `,
+      errors: [{ messageId: 'extraValue' }],
+    },
+    {
+      code: `
+interface Foo {
+  a:  (b)
+}
+      `,
+      output: `
+interface Foo {
+  a: (b)
+}
+      `,
+      errors: [{ messageId: 'extraValue' }],
+    },
+    {
+      code: `
+class Foo {
+  a:  /** comment */ b
+}
+      `,
+      output: `
+class Foo {
+  a: /** comment */ b
+}
+      `,
+      errors: [{ messageId: 'extraValue' }],
+    },
+    {
+      code: `
+class Foo {
+  a:    (     b)
+}
+      `,
+      output: `
+class Foo {
+  a: (     b)
+}
+      `,
+      errors: [{ messageId: 'extraValue' }],
+    },
+    {
+      code: `
+interface X { a:(number); };
+      `,
+      output: `
+interface X { a : (number); };
+      `,
+      options: [
+        {
+          singleLine: { beforeColon: true, afterColon: true, mode: 'strict' },
+          multiLine: { beforeColon: true, afterColon: true },
+        },
+      ],
+      errors: [{ messageId: 'missingKey' }, { messageId: 'missingValue' }],
+    },
+    {
+      code: `
+interface X { a:/** comment */ number; };
+      `,
+      output: `
+interface X { a : /** comment */ number; };
+      `,
+      options: [
+        {
+          singleLine: { beforeColon: true, afterColon: true, mode: 'strict' },
+          multiLine: { beforeColon: true, afterColon: true },
+        },
+      ],
+      errors: [{ messageId: 'missingKey' }, { messageId: 'missingValue' }],
+    },
```

Comment:
[Testing] The reported issue mentions `(string | number)`. Could you add in a cursory bit of testing for that please?

Even if the type being a union doesn't impact the formatting logic touched in this PR, I think it'd be a good idea just in case.

Reply:
Thanks, this makes sense! I added cases with `(string | number)`